cirq/contrib/routing/router_test.py

```diff
@@ -87,3 +87,36 @@ def test_router_bad_args():
         cirq.CZ(cirq.LineQubit(i), cirq.LineQubit(i + 1)) for i in range(5))
     with pytest.raises(ValueError):
         ccr.route_circuit(circuit, device_graph, algo_name=algo_name)
+
+
+def test_fail_when_operating_on_unmapped_qubits():
+    a, b, t = cirq.LineQubit.range(3)
+    swap = cirq.contrib.acquaintance.SwapPermutationGate()
+
+    # Works before swap.
+    swap_network = cirq.contrib.routing.SwapNetwork(cirq.Circuit(cirq.CZ(a, t)),
+                                                    {
+                                                        a: a,
+                                                        b: b
+                                                    })
+    with pytest.raises(ValueError, match='acts on unmapped qubit'):
+        _ = list(swap_network.get_logical_operations())
```

Comment:
Is the mock assignment for pylint?

Reply:
Yes, sometimes pylint complains that the result of an expression is unused. So I got into the habit of assigning to `_` as a way of saying "I know I'm throwing this away".

Reply:
Maybe
```
for op in swap_network.get_logical_operations():
    pass # coverage: ignore
```

It's fine as-is if you have a preference.